experiment/analysis/comparison-of-bfs-with-dfs.md

```diff
@@ -1,39 +1,43 @@
   -  Memory space is efficiently utilized in DFS while space utilization in BFS is not effective.
   -  BFS is an optimal algorithm while DFS is not optimal.
-  -  DFS constructs narrow and long trees whereas BFS constructs wide and short tree.
+  -  DFS constructs narrow and long trees whereas BFS constructs wide and short trees.
 
-### BFS traversal on Example graph
-  -  We have a graph whose vertices are A, B, C, D, E, F, G. Considering A as starting point, the steps involved in the process are:
-  - Vertex A is expanded and stored in the queue.
-  -  Vertices B, D and G successors of A, are expanded and stored in the queue. Meanwhile Vertex A is removed.
+### BFS traversal on example graph
+  -  We have a graph whose vertices are A, B, C, D, E, F, G. Considering A as the starting point, the steps involved in the process are:
+  -  Vertex A is expanded and stored in the queue.
+  -  Vertices B, D, and G, as successors of A, are expanded and stored in the queue. Meanwhile, Vertex A is removed.
   -  Now B at the front end of the queue is removed along with storing its successor vertices E and F.
-  -  Vertex D at the front end of the queue is removed, and its connected node F is already visited.
-  -  Vertex G is removed from the queue, and it has successor E which is already visited.
+  -  Vertex D at the front end of the queue is removed, and its connected node F has already been visited.
+  -  Vertex G is removed from the queue, and it has successor E which has already been visited.
   -  Now E and F are removed from the queue, and its successor vertex C is traversed and stored in the queue.
-  -  At last C is also removed and the queue is empty which means we are done.
+  -  At last C is also removed and the queue is now empty which means we are done.
   -  The generated Output is – A, B, D, G, E, F, C.
 
-### DFS traversal on Example graph
+  <img src="images/bfs.png"/>
 
-Similar to BFS lets take the same graph for performing DFS operations, and the involved steps are:
+### DFS traversal on example graph
+
+Similar to BFS lets take the same graph for performing DFS operations. The steps involved in the process are:
 
    - Considering A as the starting vertex which is explored and stored in the stack.
    - B, the successor vertex of A, is stored in the stack.
-   - Vertex B have two successors E and F, among them alphabetically E is explored first and stored in the stack.
+   - Vertex B has two successors E and F, among them alphabetically E is explored first and stored in the stack.
    - The successor of vertex E, i.e., G is stored in the stack.
    - Vertex G has two connected vertices, and both are already visited, so G is popped out from the stack.
    - Similarly, E is also removed.
-   - Now, vertex B is at the top of the stack, its another node(vertex) F is explored and stored in the stack.
+   - Now vertex B is at the top of the stack, its other node(vertex) F is explored and stored in the stack.
```

Comment:
Should we say, 'its other successor' in place of 'its other node' so we stay consistent with line 30 on the terminology?

Reply:
I agree that that is better than 'its other node'.